staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -102,10 +101,17 @@ func ObjectReaction(tracker ObjectTracker) ReactionFunc {
 			if action.GetSubresource() == "" {
 				err = tracker.Create(gvr, action.GetObject(), ns)
 			} else {
-				// TODO: Currently we're handling subresource creation as an update
-				// on the enclosing resource. This works for some subresources but
-				// might not be generic enough.
-				err = tracker.Update(gvr, action.GetObject(), ns)
+				oldObj, getOldObjErr := tracker.Get(gvr, ns, objMeta.GetName())
+				if getOldObjErr != nil {
+					return true, nil, err
+				}
+				// Check if the type has changed
+				if fmt.Sprintf("%T", oldObj) == fmt.Sprintf("%T", action.GetObject()) {
+					// TODO: Currently we're handling subresource creation as an update
+					// on the enclosing resource. This works for some subresources but
+					// might not be generic enough.
+					err = tracker.Update(gvr, action.GetObject(), ns)
+				}
```

Comment:
we need to make sure we return the object submitted, so add an else branch with this for the subresource case where we don't persist the submitted object in the tracker:

```suggestion
				} else {
					return true, action.GetObject(), nil
				}
```

Reply:
received, I add an else branch to it